File: examples-classic/messaging/aws-py/child/aws/fn.py
import json
from typing import Any, Optional, Dict
from urllib import request, error


def send_message(
    message: Any, receiving_invocation: Optional[int], context: Dict[str, Any]
) -> Any:
    url = context["workflowMessageUrl"]
    if receiving_invocation is not None:
        url += f"/{receiving_invocation}"
    headers = {
        "Content-Type": "application/json",
        "Authorization": context["authorizationHeader"],
    }
    data = json.dumps(message).encode("utf-8")
    req = request.Request(url, data=data, headers=headers, method="POST")
    try:
        with request.urlopen(req) as resp:
            return json.load(resp)
    except error.HTTPError as e:
        raise Exception(f"HTTP error: {e.code} {e.reason}") from e


def send_message_to_parent(message: Any, context: Dict[str, Any]) -> Any:
    parent_id = context.get("parentId")
    if parent_id is None:
        raise Exception("No parent to send message to")
    return send_message(message, parent_id, context)


def send_message_to_workflow(message: Any, context: Dict[str, Any]) -> Any:
    return send_message(message, None, context)

File: examples-classic/flexi-consensus-kn/participant/kn/src/fn.ts
export async function sendMessage(
  message: any,
  receivingInvocation: number | null,
  context: any,
): Promise<any> {
  let url = context.workflowMessageUrl;
  if (receivingInvocation !== null) {
    url += `/${receivingInvocation}`;
  }
  const response = await fetch(
    url,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Authorization: context.authorizationHeader,
      },
      body: JSON.stringify(message),
    },
  );
  return await response.json();
}

export async function sendMessageToParent(
  message: any,
  context: any,
): Promise<any> {
  const parentId = context.parentId;
  if (parentId === null) {
    throw new Error("No parent to send message to");
  }
  return await sendMessage(message, parentId, context);
}

export async function sendMessageToWorkflow(
  message: any,
  context: any,
): Promise<any> {
  return await sendMessage(message, null, context);
}
